pkg/ext-proc/backend/llmserverpool_reconciler.go

```diff
@@ -21,39 +36,167 @@ const (
-	Namespace      string
-	Datastore      *K8sDatastore
-	Zone           string
+	//Scheme         *runtime.Scheme
+	Record            record.EventRecorder
+	ServerPoolName    string
+	Namespace         string
+	Datastore         *K8sDatastore
+	Zone              string
+	v1alpha1clientset clientset.Interface
+	poolLister        listers.LLMServerPoolLister
+	poolSynced        cache.InformerSynced
+
+	workqueue workqueue.TypedRateLimitingInterface[cache.ObjectName]
+	// Still uses recorder
+}
+
+func NewLLMServerPoolReconciler(
+	//Params:
+	ctx context.Context,
+	serverPoolInformer informers.LLMServerPoolInformer,
+	clientset clientset.Interface,
+	kubeclientset kubernetes.Interface,
+	scheme *runtime.Scheme,
+	serverpoolName string,
+	namespace string,
+	zone string,
+	datastore *K8sDatastore,
+	//Return val:
+) *LLMServerPoolReconciler {
+	// Body:
+	//utilruntime.Must(v1alpha1scheme.AddToScheme(scheme.Scheme))
+
+	klog.V(4).Info("Creating event broadcaster")
+	eventBroadcaster := record.NewBroadcaster(record.WithContext(ctx))
+	eventBroadcaster.StartStructuredLogging(0)
+	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: kubeclientset.CoreV1().Events("")})
+	recorder := eventBroadcaster.NewRecorder(scheme, corev1.EventSource{Component: "FIGUREOUT NAMING HERE"})
+	ratelimiter := workqueue.NewTypedMaxOfRateLimiter(
+		workqueue.NewTypedItemExponentialFailureRateLimiter[cache.ObjectName](5*time.Millisecond, 1000*time.Second),
+		&workqueue.TypedBucketRateLimiter[cache.ObjectName]{Limiter: rate.NewLimiter(rate.Limit(50), 300)},
+	)
+
+	reconciler := &LLMServerPoolReconciler{
+		v1alpha1clientset: clientset,
+		poolLister:        serverPoolInformer.Lister(),
+		poolSynced:        serverPoolInformer.Informer().HasSynced,
+		Record:            recorder,
+		workqueue:         workqueue.NewTypedRateLimitingQueue(ratelimiter),
+		ServerPoolName:    serverpoolName,
+		Namespace:         namespace,
+		Zone:              zone,
+		Datastore:         datastore,
+	}
+	klog.Info("Setting up event handlers")
+
+	serverPoolInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: reconciler.enqueuePool,
+		UpdateFunc: func(old, new interface{}) {
+			reconciler.enqueuePool(new)
+		},
+		DeleteFunc: reconciler.enqueuePool,
+	})
+
+	return reconciler
+}
+
+func (r *LLMServerPoolReconciler) Run(ctx context.Context, workers int) error {
+	defer utilruntime.HandleCrash()
+	defer r.workqueue.ShutDown()
+
+	klog.Info("Starting Pool Reconciler")
+
+	klog.Info("Waiting for informer caches to sync")
+	if ok := cache.WaitForCacheSync(ctx.Done(), r.poolSynced); !ok {
+		return fmt.Errorf("failed to wait for caches to sync")
+	}
+
+	klog.Info("Starting workers ", "count ", workers)
+	// Launch two workers to process Foo resources
+	for i := 0; i < workers; i++ {
+		go wait.UntilWithContext(ctx, r.runWorker, time.Second)
+	}
+
+	klog.Info("Started workers")
+	<-ctx.Done()
+	klog.Info("Shutting down workers")
+
+	return nil
+}
```

Comment:
same thing, this function can be placed in a pkg that can be used across reconcilers. The function would accept some config struct that includes the cache sync and worker run functions.